controller/lib/hal/hal.h

```diff
@@ -52,13 +52,16 @@ class HalApi {
   // millis().
   void delay(uint32_t ms);
 
+	int analogRead(int pin);
+
   // TODO: Need at least one HAL_MOCK_METHOD.
 
 private:
 #ifdef TEST_MODE
   // Instance variables used when mocking HAL.
 
   uint32_t millis_ = 0;
+	int pin_values_[15] = {0};
```

Comment:
nit, add a TODO stating your intentions for this.  (I think you're going to add a test-only method that sets each relevant analog pin?)

I wonder whether analogRead should be a mock method or a fake.  Sort of feels like it should be a mock to me, but I guess we can just see when the time comes.

Reply:
Added a setter.